src/coreclr/vm/bundle.cpp

```diff
@@ -30,14 +30,29 @@ const SString &BundleFileLocation::Path() const
     return Bundle::AppBundle->Path();
 }
 
-Bundle::Bundle(LPCSTR bundlePath, BundleProbeFn *probe)
+#if defined(TARGET_ANDROID)
+const SString &BundleFileLocation::AppName() const
 {
-    STANDARD_VM_CONTRACT;
+    LIMITED_METHOD_CONTRACT;
 
-    _ASSERTE(probe != nullptr);
+    _ASSERTE(IsValid());
+    _ASSERTE(Bundle::AppBundle != nullptr);
 
-    m_path.SetUTF8(bundlePath);
+    return Bundle::AppBundle->Path();
+}
+#endif
+
+Bundle::Bundle(LPCSTR bundlePath, BundleProbeFn *probe, ExternalAssemblyProbeFn* externalAssemblyProbe)
+{
+    STANDARD_VM_CONTRACT;
+
+    _ASSERTE(probe != nullptr || externalAssemblyProbe != nullptr);
     m_probe = probe;
+    m_externalAssemblyProbe = externalAssemblyProbe;
+#if defined(TARGET_ANDROID)
+    m_appName.SetUTF8(bundlePath);
```

Comment:
Can we use the existing `m_path` to store the `bundlePath` for Android?

Reply:
`bundlePath` on Android isn't a path at all, it's the package (app) name.  Storing the value in `m_path` would be misleading.

Reply:
I think `m_path` with a comment about why it could be misleading would achieve the same result.

Reply:
> bundlePath on Android isn't a path at all, it's the package (app) name. Storing the value in m_path would be misleading.

It is about as misleading as passing it in argument named `bundlePath`.